src/components/dashboard/reserve/appointmentsList.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getFurnitures } from '../../../store/furnitureSlice';
+import {
+  deleteAppointment,
+  fetchAppointments,
+} from '../../../store/appointmentsSlice';
+
+function AppointmentsList() {
+  const dispatch = useDispatch();
+  const { appointments, status } = useSelector((state) => state.appointments);
+  const error = useSelector((state) => state.appointments.error);
+  const { furnitures, message } = useSelector((state) => state.furniture);
+
+  useEffect(() => {
+    // Fetch user appointments when the component mounts
+    dispatch(fetchAppointments());
+    dispatch(getFurnitures());
+  }, [dispatch, message]);
```

Comment:
- [ ] When I try to go to "My appointments", the app and the server get into an endless loop of requests that never stop:

![image](https://github.com/anbehindY/furniture-mall-frontend/assets/57421823/81eb8729-e272-4cb6-893e-5723164d9390)


The only way to stop it is to reload the page or close the app.

Kindly, make it so "My appointments" makes only one request.

_This was tested both on Chrome and firefox._

Reply:
Fixed